### ISSUE TITLE
Update timestamp links to go to the user's own-instance version of the post, to match vanilla behaviour

### ISSUE THREAD

**Issue by averixus** [glitch-soc/mastodon#2082]
### Pitch

Vanilla recently updated the timestamp links in the timeline so that they lead to the post within the user's own instance, rather than the original on the poster's instance. (This issue: https://github.com/mastodon/mastodon/issues/19652)

My request is for glitch to do the same, so that opening the timestamp link of a post in a new tab will keep the user within their own instance in that new tab.

Failing that (if glitch is set on using the original-instance link for timestamps), then my request is to add an own-instance link in the dots menu for convenience.

### Motivation

Opening a post in a new tab while browsing is useful, for example to check back on later without losing your current place in a timeline or thread. The current behaviour means that opening in a new tab brings you to a different instance where you are no longer able to interact with the post. Currently the only ways to get the own-instance version of a post open in a new tab would be either:
1. Open it in the current tab (losing track of the current thread/timeline), copy the URL, open a new tab, paste it there, return to original tab, and go back to where you were in the thread/timeline. Or,
2. Copy the original-instance link, open a new tab, navigate to your own instance, paste the URL into the search, and open the result.

Which are both very roundabout and inconvenient.

**Comment by Cassolotl** [comment]
I was also annoyed about not being able to open posts in a new tab, but then I opened this on the vanilla Mastodon issue list: *Command-clicking a post should open it and its thread in a new tab #21553*

**Comment by Plastikmensch** [comment]
I'm against this.
Trapping users on their own instance is a huge annoyance I have with vanilla.
When I click the timestamp to open it in a new tab, I want and expect it to open it in the original instance.

I see the use case though, but it breaks the advanced UI, as every time you open a toot in a new tab with the vanilla behaviour it loads all columns needlessly.
A compromise would be to have a setting which toggles between vanilla and glitch behaviour.

**Comment by averixus** [comment]
Yeah, a user-defined setting would be ideal to please everyone. I've added that to my suggestion explictly 👍

**Comment by ClearlyClaire** [comment]
I'm against this change, however I have another suggestion: currently, middle-clicking or ctrl+clicking the post itself does nothing, however it could open the post in a new tab. Is that something you'd find useful?

(I wanted to do that for both upstream and glitch-soc but haven't got around to doing that yet. Ideally it should closely match the browser's link clicking behavior, but that can depend on browser/OS, so I'm not 100% sure how to go about that)

**Comment by Plastikmensch** [comment]
> I'm against this change, however I have another suggestion: currently, middle-clicking or ctrl+clicking the post itself does nothing, however it could open the post in a new tab. Is that something you'd find useful?
> 
> (I wanted to do that for both upstream and glitch-soc but haven't got around to doing that yet. Ideally it should closely match the browser's link clicking behavior, but that can depend on browser/OS, so I'm not 100% sure how to go about that)

I also thought about that and I think it's a good solution, but one issue I see is that people want different default behaviours. Some people want just clicking to open it on the own instance, others want it to open it on the original instance.

**Comment by ClearlyClaire** [comment]
> I also thought about that and I think it's a good solution, but one issue I see is that people want different default behaviours. Some people want just clicking to open it on the own instance, others want it to open it on the original instance.

My proposal is that clicking the post itself (not the timestamp) continues opening the post in the local instance, with the addition that ctrl+clicking or middle-clicking opens the post in the local instance in a new tab. That seems consistent with the current behavior and provide the ability to quickly open a local view of the post in a new tab. The timestamp link would remain unchanged and link to the post's original instance.

**Comment by Pauliehedron** [comment]
Note - I only use the non-advanced interface:

Control click / middle click on a web page isn't very intuitive (I haven't done it since I was using Motif) and also not touchable for mobile web/PWA.  

The three dots menu on a post already lists an open original page option today.  Having an open a new tab on-instance would be helpful to me as PWA doesn't have a tab interface in non-advanced mode.

**Comment by awebeer256** [comment]
Let's broaden the scope a bit.

There are a few different kinds of things one can click on:

1. The body of a post
2. A post's timestamp
3. A person's name at the top of one of their posts.
4. Mentions, hashtags, and links to posts—each of which can exist both in posts and in someone's bio.

Each of these can be either regular-clicked or ctrl/middle(/command?)-clicked.
Each link target can be opened in the same tab or a new tab.
Each link target can be opened in the context of the current instance or the original instance.

So what behaviour do we want all of these to have? Here's my proposal:

- Post body
  - Regular: current instance, current tab
  - Mod: current instance, new tab
- Post timestamp
  - Regular: original instance, current tab
  - Mod: original instance, new tab
- Everything else
  - Regular: current instance, current tab
  - Mod: ??? instance (user-configurable), new tab

Reasoning:

1. In general, regular clicking should open the target in both the current instance and the current tab.
2. Modified clicking should open the target in a new tab.
3. In an exception to point 1, since posts have both a body and a timestamp to click on, we can split things up: the body for the current instance, and the timestamp for the original instance.

**Comment by awebeer256** [comment]
Addendum: at the top of a post, where the author is displayed, there are three different things you can click on in order to go to their user page: their display name, avatar, and username.

So we can split this up like with a post's body and timestamp: click the display name or avatar to stay in the current instance, or the username to go to the original instance. The reason is that the username explicitly states the original instance (if it's different than the current one), so it makes more sense that you would go to that instance by clicking it.

(And again, regular click to stay in the current tab; modified click to open a new tab.)
